internal/webhook/v1beta1/glanceapi_webhook.go

```diff
@@ -0,0 +1,131 @@
+	glanceapilog.Info("Defaulting for GlanceAPI", "name", glanceapi.GetName())
+
+	if glanceapi.Spec.ContainerImage == "" {
+		glanceapi.Spec.ContainerImage = glancev1beta1.GetGlanceDefaults().ContainerImageURL
+	}
+
+	return nil
+}
+
+// TODO(user): change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
+// NOTE: The 'path' attribute must follow a specific pattern and should not be modified directly here.
+// Modifying the path for an invalid path can cause API server errors; failing to locate the webhook.
+// +kubebuilder:webhook:path=/validate-glance-openstack-org-v1beta1-glanceapi,mutating=false,failurePolicy=fail,sideEffects=None,groups=glance.openstack.org,resources=glanceapis,verbs=create;update,versions=v1beta1,name=vglanceapi-v1beta1.kb.io,admissionReviewVersions=v1
+
+// GlanceAPICustomValidator struct is responsible for validating the GlanceAPI resource
+// when it is created, updated, or deleted.
+//
+// NOTE: The +kubebuilder:object:generate=false marker prevents controller-gen from generating DeepCopy methods,
+// as this struct is used only for temporary operations and does not need to be deeply copied.
+type GlanceAPICustomValidator struct {
+	// TODO(user): Add more fields as needed for validation
+}
+
+var _ webhook.CustomValidator = &GlanceAPICustomValidator{}
+
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type GlanceAPI.
+func (v *GlanceAPICustomValidator) ValidateCreate(_ context.Context, obj runtime.Object) (admission.Warnings, error) {
+	glanceapi, ok := obj.(*glancev1beta1.GlanceAPI)
+	if !ok {
+		return nil, fmt.Errorf("%w: got %T", errUnexpectedGlanceAPIObjectType, obj)
+	}
+	glanceapilog.Info("Validation for GlanceAPI upon creation", "name", glanceapi.GetName())
+
+	// TODO(user): fill in your validation logic upon object creation.
+
+	return nil, nil
+}
+
+// ValidateUpdate implements webhook.CustomValidator so a webhook will be registered for the type GlanceAPI.
+func (v *GlanceAPICustomValidator) ValidateUpdate(_ context.Context, _, newObj runtime.Object) (admission.Warnings, error) {
+	glanceapi, ok := newObj.(*glancev1beta1.GlanceAPI)
+	if !ok {
+		return nil, fmt.Errorf("%w: got %T", errUnexpectedGlanceAPIObjectType, newObj)
+	}
+	glanceapilog.Info("Validation for GlanceAPI upon update", "name", glanceapi.GetName())
+
+	// TODO(user): fill in your validation logic upon object update.
+
+	return nil, nil
+}
+
+// ValidateDelete implements webhook.CustomValidator so a webhook will be registered for the type GlanceAPI.
+func (v *GlanceAPICustomValidator) ValidateDelete(_ context.Context, obj runtime.Object) (admission.Warnings, error) {
+	glanceapi, ok := obj.(*glancev1beta1.GlanceAPI)
+	if !ok {
+		return nil, fmt.Errorf("%w: got %T", errUnexpectedGlanceAPIObjectType, obj)
+	}
+	glanceapilog.Info("Validation for GlanceAPI upon deletion", "name", glanceapi.GetName())
+
+	// TODO(user): fill in your validation logic upon object deletion.
+
```

Comment:
same as above, there is ValidateDelete(), not really used, but might be good for future use?